presentation/src/main/java/me/androidbox/presentation/login/viewmodel/SplashScreenViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package me.androidbox.presentation.login.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import me.androidbox.domain.authentication.ResponseState
+import me.androidbox.domain.authentication.usecase.AuthenticateUserUseCase
+import javax.inject.Inject
+
+@HiltViewModel
+class SplashScreenViewModel @Inject constructor(
```

Comment:
I wouldn't call it SplashScreenViewModel, since authentication of a user isn't related to the splashscreen

I would call it MainViewModel or something similar. So it's more related to the MainActivity (in this case, HomeActivity)